exporters/tests/test_deserialization.py

```diff
@@ -58,6 +60,16 @@ class Nested:
     assert actual.nested_int == 2
 
 
+def test_multi_nested():
+    @define
+    class NestedNested:
+        nested_int: int = field(metadata=nested(["foo", "com.example.int"]))
```

Comment:
I think I did not understand how this work. changing to `["foo", "com", "example.int"]` should not work as well?

Reply:
That would mean something different!

I should add a doctest-style example to `get_nested` and `nested`, but here's the short version:

```python
x: SomeDeeplyNestedDict
assert get_nested(x, "foo.com.example.int") == x["foo"]["com"]["example"]["int"]
assert get_nested(x, ["foo", "com.example.int"]) == x["foo"]["com.example.int"]
```

The first form is a convenient way to replace what we used to have, something like `build_config.spec.source.git.uri`, with `get_nested(build_config, "spec.source.git.uri")`. But when looking at labels and annotations in OpenShift, they commonly have dots in them-- and users can configure what they are, too. So we need the other form to handle that.